src/sage/rings/lazy_series_ring.py

```diff
@@ -655,11 +657,413 @@ def undefined(self, valuation=None):
+
+            sage: L.<z> = LazyPowerSeriesRing(RR)
+            sage: G = L.undefined(0)
+            sage: L.define_implicitly([(G, [log(2)])], [diff(G) - exp(-G(-z))])
+            sage: G
+            0.693147180559945 + 1.00000000000000*z + 0.500000000000000*z^2
+             - 0.0833333333333333*z^4 + 0.0222222222222222*z^6 + O(1.00000000000000*z^7)
+
+        We solve the recurrence relation in (3.12) of Prellberg and Brak
+        :doi:`10.1007/BF02183685`::
+
+            sage: q, y = QQ['q,y'].fraction_field().gens()
+            sage: L.<x> = LazyPowerSeriesRing(q.parent())
+            sage: R = L.undefined()
+            sage: L.define_implicitly([(R, [0])], [(1-q*x)*R - (y*q*x+y)*R(q*x) - q*x*R*R(q*x) - x*y*q])
+            sage: R[0]
+            0
+            sage: R[1]
+            (-q*y)/(q*y - 1)
+            sage: R[2]
+            (q^3*y^2 + q^2*y)/(q^3*y^2 - q^2*y - q*y + 1)
+            sage: R[3].factor()
+            (-1) * y * q^3 * (q*y - 1)^-2 * (q^2*y - 1)^-1 * (q^3*y - 1)^-1
+             * (q^4*y^3 + q^3*y^2 + q^2*y^2 - q^2*y - q*y - 1)
+
+            sage: Rp = L.undefined(1)
+            sage: L.define_implicitly([Rp], [(y*q*x+y)*Rp(q*x) + q*x*Rp*Rp(q*x) + x*y*q - (1-q*x)*Rp])
+            sage: all(R[n] == Rp[n] for n in range(7))
+            True
+
+        Another example::
+
+            sage: L.<z> = LazyPowerSeriesRing(QQ["x,y,f1,f2"].fraction_field())
+            sage: L.base_ring().inject_variables()
+            Defining x, y, f1, f2
+            sage: F = L.undefined()
+            sage: L.define_implicitly([(F, [0, f1, f2])], [F(2*z) - (1+exp(x*z)+exp(y*z))*F - exp((x+y)*z)*F(-z)])
+            sage: F
+            f1*z + f2*z^2 + ((-1/6*x*y*f1+1/3*x*f2+1/3*y*f2)*z^3)
+             + ((-1/24*x^2*y*f1-1/24*x*y^2*f1+1/12*x^2*f2+1/12*x*y*f2+1/12*y^2*f2)*z^4)
+             + ... + O(z^8)
+            sage: sol = 1/(x-y)*((2*f2-y*f1)*(exp(x*z)-1)/x - (2*f2-x*f1)*(exp(y*z)-1)/y)
+            sage: F - sol
+            O(z^7)
+
+        We need to specify the initial values for the degree 1 and 2
+        components to get a unique solution in the previous example::
+
+            sage: L.<z> = LazyPowerSeriesRing(QQ['x','y','f1'].fraction_field())
+            sage: L.base_ring().inject_variables()
+            Defining x, y, f1
+            sage: F = L.undefined()
+            sage: L.define_implicitly([F], [F(2*z) - (1+exp(x*z)+exp(y*z))*F - exp((x+y)*z)*F(-z)])
+            sage: F
+            <repr(...) failed: ValueError: could not determine any coefficients:
+                coefficient [3]: 6*series[3] + (-2*x - 2*y)*series[2] + (x*y)*series[1] == 0>
+
+            sage: F = L.undefined()
+            sage: L.define_implicitly([(F, [0, f1])], [F(2*z) - (1+exp(x*z)+exp(y*z))*F - exp((x+y)*z)*F(-z)])
+            sage: F  # random
+            <repr(...) failed: ValueError: could not determine any coefficients:
+                coefficient [3]: 6*series[3] + (-2*x - 2*y)*series[2] + (x*y*f1) == 0>
```

Comment:
```suggestion
        The following test results in a failure, but at different points
        on different machines::

            mac: coefficient [3]: 6*series[3] + (-2*x - 2*y)*series[2] + (x*y*f1) == 0
            linux: ...
        
        Due to this, we only test that the base output::

            sage: F  # random
            <repr(...) failed: ValueError: could not determine any coefficients:
                coefficient ... == 0>
```
I don't think it is good to mark this as `# random` since that then only checks that an (uncaught) error is not raised. Instead, I think we should be more specific, both in the test and in the explanation.

Reply:
I don't mind, but did you intend to remove#random in 812?

Reply:
Ah, yes, I meant to remove that.

Reply:
Oh, and it is only on some MacOS versions

Reply:
Oh, and it is only on some MacOS versions

Reply:
Actually, I think that "but at different points on different machines::" might be misunderstood.  The only thing that may differ is the order in which the monomials in the polynomial "6*series[3] + (-2*x - 2*y)*series[2] + (x*y*f1)" appear.  So here is another suggestion:

```suggestion
        We have thus specified that the degree 1 component equals `f1`.  This is still
        not enough to remove the ambiguity, so an error is raised.  However, we can
        see that the dependence on `series[1]` disappeared.  The equation which has
        no unique solution is now
        `6*series[3] + (-2*x - 2*y)*series[2] + (x*y*f1) == 0`::

            sage: F  # random
            <repr(...) failed: ValueError: could not determine any coefficients:
                coefficient [3]: ... == 0>
```

Reply:
If you're removing the `# random`, then okay. Although it would be good to explain a bit why the `...` is in the doctest here. Can you give the two different outputs, at least here on this PR for the record?

Also, it would be {{{``}}} on both sides for code formatting.